samples/openshift-cloud-terminal.yaml

```diff
@@ -0,0 +1,26 @@
+apiVersion: workspace.che.eclipse.org/v1alpha1
+kind: Workspace
+metadata:
+  name: openshift-terminal
+  annotations:
+    org.eclipse.che.workspace/immutable: "true"
+  labels:
+    console.openshift.io/cloudshell: "true"
+spec:
+  started: true
+  devfile:
+    apiVersion: 1.0.0
+    metadata:
+      name: cloud-shell
```

Comment:
I see you have cloud-shell and openshift-cloud-terminal mixed together here. 

Also is in intentional that you're using codeready-workspaces containers in *gasp* the upstream project? :) I'm OK with that but... does that also introduce a circular dependency?

Reply:
it's better to change this particular cloud-shell word. Will do.

Apart from that there is still cloud-shell cheEditor plugin in internal registry which I did not touch for backward compatibility. I think it eventually will be removed https://github.com/che-incubator/che-workspace-operator/tree/master/internal-registry/eclipse/cloud-shell/nightly

> Also is in intentional that you're using codeready-workspaces containers in gasp the upstream project? :) I'm OK with that but... does that also introduce a circular dependency?

It's an example that shows which workspace configuration OpenShift Console is supposed to create. The actual configuration is stored in OpenShift Console Repo https://github.com/openshift/console/blob/cd751e4badfd55d79df916ca8bca0b6126c446b3/frontend/packages/console-app/src/components/cloud-shell/cloud-shell-utils.ts#L76

So, I don't think it should lead to circular dependency... The thing we must release here - openshift-cloud-terminal configuration via controller image + OLM...
But I may be wrong and more questions/concerns are welcome if words above do not answer a question.

Reply:
Is anyone using the "cloud-shell cheEditor plugin" today? Does it work? If this "cloud-terminal" is the replacement for that, and the old one is a version <1.0, then it's incubating, subject to change/deletion. 

I only see a "nightly" version here: 

https://github.com/eclipse/che-plugin-registry/blob/master/v3/plugins/eclipse/cloud-shell/nightly/meta.yaml

So I think you're safe to just delete it. Unversioned "nightly" releases are throwaway builds, and are never IMHO supported. 

However I see the machine-exec nightly is ALSO affected by this old cloud-shell:

https://github.com/eclipse/che-plugin-registry/commit/74fcb49997f86a9fe4a8419292c42f8b107268c2#diff-6bd840228d47bceca757b6a33c8ad2a2

Will the machine exec continue to reference the "cloud-shell" or will it need to be updated to the new "cloud-terminal" ?

https://github.com/eclipse/che-plugin-registry/blob/master/v3/plugins/eclipse/che-machine-exec-plugin/nightly/meta.yaml#L30

Reply:
> Is anyone using the "cloud-shell cheEditor plugin" today? Does it work? If this "cloud-terminal" is the replacement for that, and the old one is a version <1.0, then it's incubating, subject to change/deletion.

Now it's used by OpenShift console https://github.com/openshift/console/blob/cd751e4badfd55d79df916ca8bca0b6126c446b3/frontend/packages/console-app/src/components/cloud-shell/cloud-shell-utils.ts#L70
Once we merge this PR and they update source code to use command-line-terminal - I think we're safe to remove cloud-shell at all.

> Will the machine exec continue to reference the "cloud-shell" or will it need to be updated to the new "cloud-terminal" ?

Actually that our front-end is not used anywhere.
Theia based workspaces - uses Theia extension.
OpenShift Console uses their own frontend.

And CloudShell Frontend in machine exec needs to be revised and eventually removed or used somewhere as separately.
And I'll create a PR into plugin-registry to stop serving CloudShell UI by che-machine-exec.